integration-tests/tests/websocket/websockets.spec.js

```diff
@@ -14,40 +14,83 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+function isSafari(browser) {
+  const UNSUPPORTED_BROWSERS = ['Safari'];
+  const currentBrowser = browser.options.desiredCapabilities.browserName;
+  return UNSUPPORTED_BROWSERS.includes(currentBrowser);
+}
+
 module.exports = {
+  afterEach : function(browser) {
+    browser.globals.clearReceivedSpans();  
+  },
   'can produce websocket events': async function(browser) {
-    const UNSUPPORTED_BROWSERS = ['Safari'];
-    const currentBrowser = browser.options.desiredCapabilities.browserName;
-    if (UNSUPPORTED_BROWSERS.includes(currentBrowser)) {
+    if (isSafari(browser)) {
       return;
     }
-
-    browser.globals.clearReceivedSpans();
     const url = browser.globals.getUrl('/websocket/websocket.ejs');
     await browser.url(url);
 
-    await browser.click('#connectWebsocketsBtn');
-    const wsConnectionSpan = await browser.globals.findSpan(span => span.name === 'connect');          
-    await browser.url(`${browser.globals.baseUrl}empty-page`);
+    await browser.click('#connectWs');
+    const wsConnectionSpan = await browser.globals.findSpan(span => span.name === 'connect', -1000);          
 
     await browser.assert.strictEqual(wsConnectionSpan.tags['location.href'], url);
     await browser.assert.strictEqual(wsConnectionSpan.tags['app'], 'splunk-otel-js-dummy-app');
     await browser.assert.strictEqual(wsConnectionSpan.tags['component'], 'websocket');
-    await browser.assert.strictEqual(wsConnectionSpan.tags['ot.status_code'], 'UNSET'); //pointless check, always OK
+    await browser.assert.strictEqual(wsConnectionSpan.tags['ot.status_code'], 'UNSET');
     await browser.assert.strictEqual(wsConnectionSpan.tags['error'], undefined);
   },
   'websocket url can be ignored': async function(browser) {
-    const UNSUPPORTED_BROWSERS = ['Safari'];
-    const currentBrowser = browser.options.desiredCapabilities.browserName;
-    if (UNSUPPORTED_BROWSERS.includes(currentBrowser)) {
+    if (isSafari(browser)) {
       return;
     }
-    
-    browser.globals.clearReceivedSpans();
     await browser.url(browser.globals.getUrl('/websocket/websocket-ignored.ejs'));
-    await browser.click('#connectWebsocketsBtn');
+    await browser.click('#connectWs');
 
     await browser.globals.findSpan(span => span.name === 'websocket-guard-span');   
     await browser.assert.not.ok(browser.globals.receivedSpans.find(span => span.name === 'connect'));
-  }
+  },
+  'sending send and on message create a span': async function(browser) {
+    if (isSafari(browser)) {
+      return;
+    }
```

Comment:
The below tests look great but can you check the other basic attributes, please?  `protocol`, `http.url`, `component===websocket` (plus `span.kind`)

Reply:
Sure